config/v1/types_console.go

```diff
@@ -26,6 +28,31 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// route contains hostname and serving certificate for replacing the default route for the console.
+	// If a custom route is specified, a new route will be created with the provided hostname.
+	// The default console route will be maintained to reserve the default hostname for console if the custom route is removed.
+	// If not specified, default route will be 	used.
+	// +optional
+	Route ConsoleConfigRoute `json:"route,omitempty"`
+}
+
+// ConsoleConfigRoute holds information on external route access to console.
+type ConsoleConfigRoute struct {
+	// name of the route to be created.
+	Name string `json:"name"`
```

Comment:
why is this configurable

Reply:
we have been pointed to https://github.com/openshift/api/blob/master/imageregistry/v1/types.go#L304-L314 so wanted to align with what we already use

Reply:
feel free to improve on what was done in imageregistry.  I don't think we had a good reason to make the route name configurable.

Reply:
We can drop the name.  
@jhadvig since we likely want to keep the default route (squat on it) in case the custom route is removed, we can just hard code `console-custom` as name for the new route.